dialer.go

```diff
@@ -213,12 +218,19 @@ func (d *Dialer) dialContext(ctx context.Context, network string, address string
 			address = net.JoinHostPort(address, port)
 		}
 	}
-	return (&net.Dialer{
+
+	dialer := &net.Dialer{
 		LocalAddr:     d.LocalAddr,
 		DualStack:     d.DualStack,
 		FallbackDelay: d.FallbackDelay,
 		KeepAlive:     d.KeepAlive,
-	}).DialContext(ctx, network, address)
+	}
+
+	if d.TLS != nil {
+		return tls.DialWithDialer(dialer, network, address, d.TLS)
```

Comment:
It’s sad that we have to choose between supporting context cancellations and TLS support. 

Do you think we could keep context cancellation by doing a basic dialer.DialContext, then call tls.Client, and tlsConn.Handshake in a separate goroutine so the main one can select in the context cancellation and completion of the handshake.

Reply:
Sure.  About to head out to the movies with family.  I'll make that change when I get back.

Reply:
No hurry :+1: